build-tools/packages/build-tools/src/repoPolicyCheck/handlers/npmPackages.ts

```diff
@@ -637,60 +622,57 @@ export const handlers: Handler[] = [
 				: undefined;
 		},
 		resolver: (file) => {
-			let json;
-			try {
-				json = JSON.parse(readFile(file));
-			} catch (err) {
-				return { resolved: false, message: "Error parsing JSON file: " + file };
-			}
+			updatePackageJsonFile(path.dirname(file), (json) => {
+				const hasScriptsField = Object.prototype.hasOwnProperty.call(json, "scripts");
```

Comment:
I think the version of node we use has the more modern https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn which was added as a nicer alternative to this. You might have to bump the typescript library target, but that should be safe. According to https://node.green/ it looks like it has been supported since nodeJS 6.4

Reply:
oops, read that wrong, looks like that needs node 16.10. So it might be ok now, but maybe isn't